packages/performance/src/services/transport_service.test.ts

```diff
@@ -78,36 +80,128 @@ describe('Firebase Performance > transport_service', () => {
     clock.tick(INITIAL_SEND_TIME_DELAY_MS);
     testTransportHandler('someEvent');
     clock.tick(DEFAULT_SEND_INTERVAL_MS);
+    await Promise.resolve();
     expect(fetchStub).to.have.been.calledOnce;
   });
 
   it('successful send a meesage to transport', () => {
-    const transportDelayInterval = 30000;
     const setting = SettingsService.getInstance();
     const flTransportFullUrl =
       setting.flTransportEndpointUrl + '?key=' + setting.transportKey;
     fetchStub.withArgs(flTransportFullUrl, match.any).resolves(
       // DELETE_REQUEST means event dispatch is successful.
-      new Response(
-        '{\
-        "nextRequestWaitMillis": "' +
-          transportDelayInterval +
-          '",\
-        "logResponseDetails": [\
-          {\
-            "responseAction": "DELETE_REQUEST"\
-          }\
-        ]\
-      }',
-        {
-          status: 200,
-          headers: { 'Content-type': 'application/json' }
-        }
-      )
+      generateSuccessResponse()
     );
 
     testTransportHandler('event1');
     clock.tick(INITIAL_SEND_TIME_DELAY_MS);
     expect(fetchStub).to.have.been.calledOnce;
   });
+
+  it('sends up to the maximum event limit in one request', async () => {
```

Comment:
This unit test is hard to follow. Can we instead have one in which there is more than 1000 logs in the queue and check the number of logs received by fetch stub to be equal to 1000?

Reply:
Simplified unit test to capture only the first dispatch to be equal to 1000 events. Since we also have the splice logic of keeping the remaining events in the queue, added validation on remaining event count is kept.